juriscraper/pacer/free_documents.py

```diff
@@ -93,8 +95,17 @@ def query(self, start, end, sort="date_filed", day_span=7):
             }
             response = self.session.post(f"{self.url}?{nonce}", data=data)
             responses.append(response)
+            responses_with_params.append(
+                {
+                    "response": response,
+                    "start": _start,
+                    "end": _end,
+                    "court_id": self.court_id,
+                }
+            )
```

Comment:
Hm, I had a hunch that you could get this same information from the response object, and you kind of can:

```
from urllib.parse import parse_qsl

params = dict(parse_qsl(response.request.body))
filename = free_opinions_report_{report.court_id}_from_{params.start_date}_to_{params.end_date}.html
```

Somehow that feels better to me than passing extra data around in a list that we already have?

Reply:
@mlissner I'm going to defer to your judgment here @mlissner

Reply:
I think if the above works, we should just do it that way, but per my usual, my code is just notional, and I guess it'd be nice to confirm it works before closing this one.

Reply:
I had already tried this approach but it didn't work that way, that is why i modified juriscraper but I did a little more research, because it is a post method the data is in the multipart/form-data form and is returned as bytes and we can get the params from the request body but using something different.

We have two options to obtain the request parameters from the response:

Using the requests_toolbelt library which would have to be added to courtlistener, this would allow us to parse the body of the request.

```
from requests_toolbelt.multipart import decoder

multipart_data = decoder.MultipartDecoder(body_bytes,
                                          response.request.headers['Content-Type'])

params_dict = {}
for part in multipart_data.parts:
    content_disposition = part.headers[b'Content-Disposition'].decode('utf-8')
    name = content_disposition.split(';')[1].split('=')[1].strip('"')
    params_dict[name] = part.text

print("Body params:", params_dict)
```


The second option does not require a library, just good python.

```
body_bytes = response.request.body
body_str = body_bytes.decode('utf-8')

content_type = response.request.headers['Content-Type']
boundary = content_type.split("boundary=")[-1]

parts = body_str.split(f'--{boundary}')

params_dict = {}
for part in parts:
    if 'Content-Disposition' in part:
        name_index = part.find('name="') + len('name="')
        name_end_index = part.find('"', name_index)
        name = part[name_index:name_end_index]

        value_index = part.find('\r\n\r\n') + len('\r\n\r\n')
        value_end_index = part.find('\r\n', value_index)
        value = part[value_index:value_end_index]

        params_dict[name] = value
        
print("Body params:", params_dict)

```

I believe we can use either of these two options and thus not modify juriscraper

Reply:
Oh wow, yeah, that's not pretty. Let's not do that and let's go with plan A: Your way!

Reply:
@mlissner Can I merge this PR now?